python/cudf/cudf/core/multiindex.py

```diff
@@ -1636,72 +1638,67 @@ def _level_index_from_level(self, level):
-            - If index is unique, search result is unique, return a single int.
-            - If index is monotonic, index is returned as a slice object.
-            - Otherwise, cudf attempts a best effort to convert the search
-              result into a slice object, and will return a boolean mask if
-              failed to do so. Notice this can deviate from Pandas behavior
-              in some situations.
-
-        Examples
-        --------
-        >>> import cudf
-        >>> mi = cudf.MultiIndex.from_tuples(
-        ...     [('a', 'd'), ('b', 'e'), ('b', 'f')])
-        >>> mi.get_loc('b')
-        slice(1, 3, None)
-        >>> mi.get_loc(('b', 'e'))
-        1
-        >>> non_monotonic_non_unique_idx = cudf.MultiIndex.from_tuples(
-        ...     [('c', 'd'), ('b', 'e'), ('a', 'f'), ('b', 'e')])
-        >>> non_monotonic_non_unique_idx.get_loc('b') # differ from pandas
-        slice(1, 4, 2)
-
-        .. pandas-compat::
-            **MultiIndex.get_loc**
-
-            The return types of this function may deviates from the
-            method provided by Pandas. If the index is neither
-            lexicographically sorted nor unique, a best effort attempt is made
-            to coerce the found indices into a slice. For example:
-
-            .. code-block::
-
-                >>> import pandas as pd
-                >>> import cudf
-                >>> x = pd.MultiIndex.from_tuples([
-                ...     (2, 1, 1), (1, 2, 3), (1, 2, 1),
-                ...     (1, 1, 1), (1, 1, 1), (2, 2, 1),
-                ... ])
-                >>> x.get_loc(1)
-                array([False,  True,  True,  True,  True, False])
-                >>> cudf.from_pandas(x).get_loc(1)
-                slice(1, 5, 1)
-        """
+    def get_indexer(self, target, method=None, limit=None, tolerance=None):
         if tolerance is not None:
             raise NotImplementedError(
                 "Parameter tolerance is not supported yet."
             )
-        if method is not None:
+        if method == "nearest":
             raise NotImplementedError(
-                "only the default get_loc method is currently supported for"
-                " MultiIndex"
+                f"{method=} is not supported yet for MultiIndex."
             )
 
+        result = cudf.core.column.full(
+            len(target),
+            fill_value=-1,
+            dtype=libcudf.types.size_type_dtype,
+        )
+        if not len(self):
+            return result.values
+        try:
+            target = cudf.MultiIndex.from_tuples(target)
+        except TypeError:
+            return result.values
+
+        lcols, rcols = map(
+            list,
+            zip(
+                *[
+                    _match_join_keys(lcol, rcol, "inner")
+                    for lcol, rcol in zip(
+                        target._data.columns, self._data.columns
+                    )
+                ]
+            ),
+        )
```

Comment:
I think the `map(list` part somewhat obfuscates that this is just the standard "transpose" pattern. Maybe move the listification to the call to `join`?

Reply:
Done, split it into two parts to make it explicit.